playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml

```diff
@@ -102,12 +112,26 @@
       state: link
     when: ca_crt_stat.stat.isreg and not ca_bundle_stat.stat.exists
 
+  # Run the upgrade hook prior to restarting services/system if defined:
+  - debug: msg="Running upgrade hook {{ openshift_master_upgrade_hook }}"
```

Comment:
I think the current state is about the best we can do but I'm open to more suggestions if people have them. We tried to come up with other naming on IRC and this was my preferred solution. I can't forsee needing more granularity right now, the only major things between "pre" and "upgrade" right now is systemd units and packages, the rest are predominantly one off upgrade tasks that will likely never run again. As it stands today I think we're pretty well covered, and we can adapt in the future if necessary. We can add new options in future and if necessary, deprecate one of these.